src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -232,51 +322,166 @@ def frobenius_charpoly(self, var='X'):
 
         ::
 
-            sage: trace = phi.frobenius_trace()
-            sage: trace
-            (4*z3^2 + 6*z3 + 3)*T + 3*z3^2 + z3 + 4
-            sage: norm = phi.frobenius_norm()
-            sage: norm
-            (5*z3^2 + 2*z3)*T^2 + (4*z3^2 + 3*z3)*T + 5*z3^2 + 2*z3
+            sage: phi.frobenius_charpoly(algorithm="NotImplemented")
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: algorithm "NotImplemented" not implemented
+
+        ALGORITHM:
+
+        By default, this method uses the so-called *crystalline*
+        algorithm which computes the characteristic polynomial of the
+        Frobenius acting on the crystalline cohomology of the Drinfeld
+        module. For further details, see [Ang1997]_.
+
+        The available options for 'algorithm' are:
+
+        - ``'crystalline'`` -- Computes the characteristic polynomial of the
+                          Frobenius endomorphism on the crystalline cohomology
+                          of a Drinfeld module.
+
+        - ``'motive'`` -- Based on computing the characteristic polynomial of
+                    the Frobenius endomorphism on the motive of a
+                    Drinfeld module. This instantiates the Frobenius
+                    as a morphism object and calls its
+                    ``'characteristic_polynomial'`` method.
```

Comment:
Thanks!

I'm however not completely sure about the formatting. I let @kryzar give his (certainly strong) opinion on this :-).

Reply:
Hihihi, thanks. It should really be this, as elsewhere in the code:

```suggestion
        - ``'crystalline'`` -- Computes the characteristic polynomial of the
          Frobenius endomorphism on the crystalline cohomology of a Drinfeld
          module.

        - ``'motive'`` -- Based on computing the characteristic polynomial of
          the Frobenius endomorphism on the motive of a Drinfeld module. This
          instantiates the Frobenius as a morphism object and calls its
          ``'characteristic_polynomial'`` method.
```